### DISCUSSION TITLE
Adding rel attribute to Url picker

### DISCUSSION THREAD

**Discussion by kfr-dwarf** [umbraco/Umbraco-CMS#10492] [Features and ideas]
### Rel attribute on url picker
I've recently had a client request  similar to the one discussed in this thread, where the client asked for the option to specify rel attributes on links for SEO purposes.
https://our.umbraco.com/forum/using-umbraco-and-getting-started/91058-umbraco-link-configurations

It would be great to see the rel attribute added on url picker with the option of providing free text values such as "noopener", "noreferrer", "nofollow" etc.

**Comment by nul800sebastiaan** [comment]
Thanks @kfr-dwarf, that's an interesting suggestion! However, maybe we can make it a bit easier even? Instead of having to bother editors with terminology they're most likely not familiar with - how about we do this automatically, when you choose a link that is not internal then we automatically add `noopener noreferrer` to it (which implicitly makes it a `nofollow` at the same time).

**Reply by kfr-dwarf** [reply]
Hi @nul800sebastiaan , thanks for the reply!
I really like that idea, as you are probably right that most editors wouldn't know what to choose for such a property anyway. I do however not know enough about the rel attribute, to tell whether it would be a good idea to always default external links to `noopener noreferrer` .

Maybe the right solution would be a mix between the two, so external links default to `noopener noreferrer`, but with the option to override for editors who do want more granular control?

**Comment by DanDiplo** [comment]
Adding `noopener` to external links automatically would make sense (it's something I always do in my builds), but not `noreferrer` - the latter should be a choice. For instance, if you link to Amazon product, you may require that `referrer` information to be passed on as part of your endorsement. 

I also don't think `noreferrer` is the same as `nofollow` - the former instructs a search engine to not pass on the referrer header whereas the latter prevents it following the link at all. I believe `noreferrer` does also implicitly mean `noopener` as well, so it covers that case. Phew!

But there are a lot of valid `rel` attributes, as you can see - https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel 

But I do agree it would be a nice option and one that advanced editors may be looking for. It just requires some thought to how it's done - whether it's a simple free text field or whether it's some form of dropdown (and how does that cope with multiple items)?

**Reply by ronaldbarendse** [reply]
Rendering links and automatically setting the `noopener` attribute would be very easy when this PR is merged: https://github.com/umbraco/Umbraco-CMS/pull/9031.

But if you need more data/properties on a link, I would suggest going for a more generic approach and use the Block List editor (maybe even with a custom backoffice view).

**Comment by maartenmensink** [comment]
Granular control would be a great feature to have in the RTE but i also can imagine one would have a default setting for all external links of be able to define a list of domains that should be automatically set to nofollow. This would allow easy handling of default behavior and give the option to overwrite default behavior where needed.

A possible way could be:

nofollow on exteral links [True|False]

nofollow on (sub)domains
- *.umbraco.net
